### PR TITLE
HDDS-2121. Create a shaded ozone filesystem (client) jar

### PR DESCRIPTION
**Pull request by elek**
We need a shaded Ozonefs jar that does not include Hadoop ecosystem components (Hadoop, HDFS, Ratis, Zookeeper).

A common expected use case for Ozone is Hadoop clients (3.2.0 and later) wanting to access Ozone via the Ozone Filesystem interface. For these clients, we want to add Ozone file system jar to the classpath, however we want to use Hadoop ecosystem dependencies that are `provided` and already expected to be in the client classpath.

Note that this is different from the legacy jar which bundles a shaded Hadoop 3.2.0.

See: https://issues.apache.org/jira/browse/HDDS-2121

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 37 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 33 | hadoop-ozone in trunk failed. |
| -1 | compile | 20 | hadoop-ozone in trunk failed. |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 950 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 158 | trunk passed |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 35 | hadoop-ozone in the patch failed. |
| -1 | compile | 26 | hadoop-ozone in the patch failed. |
| -1 | javac | 26 | hadoop-ozone in the patch failed. |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 672 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 153 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 128 | hadoop-hdds in the patch failed. |
| -1 | unit | 27 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 2520 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.ozoneimpl.TestOzoneContainer |
|   | hadoop.ozone.container.keyvalue.TestKeyValueContainer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1452 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient xml |
| uname | Linux b538116f5112 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 85b1c72 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/artifact/out/branch-compile-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/artifact/out/patch-compile-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/testReport/ |
| Max. process+thread count | 412 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozonefs-lib-current U: hadoop-ozone/ozonefs-lib-current |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 77 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 30 | hadoop-ozone in trunk failed. |
| -1 | compile | 19 | hadoop-ozone in trunk failed. |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1056 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 163 | trunk passed |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 30 | hadoop-ozone in the patch failed. |
| -1 | compile | 21 | hadoop-ozone in the patch failed. |
| -1 | javac | 21 | hadoop-ozone in the patch failed. |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 1 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 733 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 150 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 157 | hadoop-hdds in the patch failed. |
| -1 | unit | 26 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 2734 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.keyvalue.TestKeyValueContainer |
|   | hadoop.ozone.container.ozoneimpl.TestOzoneContainer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1452 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient xml |
| uname | Linux 407f30b66d7b 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 85b1c72 |
| Default Java | 1.8.0_212 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/artifact/out/branch-compile-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/artifact/out/patch-compile-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/testReport/ |
| Max. process+thread count | 306 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozonefs-lib-current U: hadoop-ozone/ozonefs-lib-current |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by arp7** [comment]
The build seems to be failing in Jenkins. However I am able to compile Ozone locally with your patch.

File: hadoop-ozone/ozonefs-lib-current/pom.xml [hunk 1]
```diff
@@ -83,6 +63,78 @@
...
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <artifactSet>
+                <excludes>
+                  <exclude>classworlds:classworlds</exclude>
```
Comment (bharatviswa504) [inline hadoop-ozone/ozonefs-lib-current/pom.xml]: What is this classworlds:classworlds, is this mistakenly added from maven example?

Comment (bharatviswa504) [reply hadoop-ozone/ozonefs-lib-current/pom.xml]: Do we need this?

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 51 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 34 | hadoop-ozone in trunk failed. |
| -1 | compile | 22 | hadoop-ozone in trunk failed. |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1150 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 178 | trunk passed |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 35 | hadoop-ozone in the patch failed. |
| -1 | compile | 24 | hadoop-ozone in the patch failed. |
| -1 | javac | 24 | hadoop-ozone in the patch failed. |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 799 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 178 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 250 | hadoop-hdds in the patch failed. |
| -1 | unit | 29 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 35 | The patch does not generate ASF License warnings. |
| | | 3069 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestSCMContainerPlacementRackAware |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1452 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient xml |
| uname | Linux 48f13474670c 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / eefe9bc |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/artifact/out/branch-compile-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/artifact/out/patch-compile-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/testReport/ |
| Max. process+thread count | 553 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozonefs-lib-current U: hadoop-ozone/ozonefs-lib-current |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/4/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 2872 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 38 | hadoop-ozone in trunk failed. |
| -1 | compile | 24 | hadoop-ozone in trunk failed. |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1210 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 168 | trunk passed |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 30 | hadoop-ozone in the patch failed. |
| -1 | compile | 22 | hadoop-ozone in the patch failed. |
| -1 | javac | 22 | hadoop-ozone in the patch failed. |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 727 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 152 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 270 | hadoop-hdds in the patch passed. |
| -1 | unit | 25 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 5807 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1452 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient xml |
| uname | Linux 629d1f742a64 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / f580a87 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/5/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/5/artifact/out/branch-compile-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/5/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/5/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/5/artifact/out/patch-compile-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/5/testReport/ |
| Max. process+thread count | 467 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozonefs-lib-current U: hadoop-ozone/ozonefs-lib-current |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/5/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/ozonefs-lib-current/pom.xml (hunk 1 above)
Comment (elek) [reply hadoop-ozone/ozonefs-lib-current/pom.xml]: No, we don't. Thanks to point to it. Removed.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 2470 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| -1 | mvninstall | 33 | hadoop-ozone in trunk failed. |
| -1 | compile | 21 | hadoop-ozone in trunk failed. |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 981 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 50 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 33 | hadoop-ozone in the patch failed. |
| -1 | compile | 25 | hadoop-ozone in the patch failed. |
| -1 | javac | 25 | hadoop-ozone in the patch failed. |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 667 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 49 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 277 | hadoop-hdds in the patch failed. |
| -1 | unit | 29 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 5053 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.placement.algorithms.TestSCMContainerPlacementRackAware |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1452 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient xml |
| uname | Linux fef328d63025 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 5db32b8 |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/branch-javadoc-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/patch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/patch-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/testReport/ |
| Max. process+thread count | 519 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozonefs-lib-current U: hadoop-ozone/ozonefs-lib-current |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1452/6/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by arp7** [comment]
I assume all the test failures are unrelated. There is one SmokeTest failure for MapReduce.

@bharatviswa504 ok to commit this?

**Comment by bharatviswa504** [comment]
`$ ./test.sh 
Removing network hadoop32_default
WARNING: Network hadoop32_default not found.
Creating network "hadoop32_default" with the default driver
Creating hadoop32_scm_1      ... done
Creating hadoop32_datanode_1 ... done
Creating hadoop32_datanode_2 ... done
Creating hadoop32_datanode_3 ... done
Creating hadoop32_nm_1       ... done
Creating hadoop32_rm_1       ... done
Creating hadoop32_s3g_1      ... done
Creating hadoop32_om_1       ... done
3 datanodes are up and registered to the scm
==============================================================================
hadoop32-createmrenv :: Create directories required for MR test               
==============================================================================
Create test volume, bucket and key                                    | PASS |
------------------------------------------------------------------------------
Create user dir for hadoop                                            | PASS |
------------------------------------------------------------------------------
hadoop32-createmrenv :: Create directories required for MR test       | PASS |
2 critical tests, 2 passed, 0 failed
2 tests total, 2 passed, 0 failed
==============================================================================
Output:  /tmp/smoketest/hadoop32/result/robot-hadoop32-hadoop32-createmrenv-scm.xml
==============================================================================
hadoop32-hadoopo3fs :: Test ozone fs with hadoopfs                            
==============================================================================
Test hadoop dfs                                                       | PASS |
------------------------------------------------------------------------------
hadoop32-hadoopo3fs :: Test ozone fs with hadoopfs                    | PASS |
1 critical test, 1 passed, 0 failed
1 test total, 1 passed, 0 failed
==============================================================================
Output:  /tmp/smoketest/hadoop32/result/robot-hadoop32-hadoop32-hadoopo3fs-rm.xml
==============================================================================
hadoop32-mapreduce :: Execute MR jobs                                         
==============================================================================
Execute PI calculation                                                | PASS |
------------------------------------------------------------------------------
Execute WordCount                                                     | PASS |
------------------------------------------------------------------------------
hadoop32-mapreduce :: Execute MR jobs                                 | PASS |
2 critical tests, 2 passed, 0 failed
2 tests total, 2 passed, 0 failed
==============================================================================
Output:  /tmp/smoketest/hadoop32/result/robot-hadoop32-hadoop32-mapreduce-rm.xml
Stopping hadoop32_s3g_1      ... done
Stopping hadoop32_datanode_1 ... done
Stopping hadoop32_nm_1       ... done
Stopping hadoop32_datanode_3 ... done
Stopping hadoop32_datanode_2 ... done
Stopping hadoop32_om_1       ... done
Stopping hadoop32_scm_1      ... done
Stopping hadoop32_rm_1       ... done
Removing hadoop32_s3g_1      ... done
Removing hadoop32_datanode_1 ... done
Removing hadoop32_nm_1       ... done
Removing hadoop32_datanode_3 ... done
Removing hadoop32_datanode_2 ... done
Removing hadoop32_om_1       ... done
Removing hadoop32_scm_1      ... done
Removing hadoop32_rm_1       ... done
Removing network hadoop32_default
Log:     /Users/bviswanadham/workspace/myozone/hadoop/hadoop-ozone/dist/target/ozone-0.5.0-SNAPSHOT/compose/ozone-mr/hadoop32/result/log.html
Report:  /Users/bviswanadham/workspace/myozone/hadoop/hadoop-ozone/dist/target/ozone-0.5.0-SNAPSHOT/compose/ozone-mr/hadoop32/result/report.html`

Ran it locally test passed, the test failure is not related to this patch. 

I will commit this to the trunk. Thank You @elek for the improvement.

**Pull request merged**